etcdmain/grpc_proxy.go

```diff
@@ -358,11 +368,26 @@ func newGRPCProxyServer(lg *zap.Logger, client *clientv3.Client) *grpc.Server {
 	electionp := grpcproxy.NewElectionProxy(client)
 	lockp := grpcproxy.NewLockProxy(client)
 
-	server := grpc.NewServer(
+	gopts := []grpc.ServerOption{
 		grpc.StreamInterceptor(grpc_prometheus.StreamServerInterceptor),
 		grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor),
 		grpc.MaxConcurrentStreams(math.MaxUint32),
-	)
+	}
+	if grpcKeepAliveMinTime > time.Duration(0) {
+		gopts = append(gopts, grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
+			MinTime:             grpcKeepAliveMinTime,
+			PermitWithoutStream: false,
+		}))
+	}
+	if grpcKeepAliveInterval > time.Duration(0) &&
```

Comment:
we probably should return an error if only of the two points are set.

Reply:
I go through the GRPC's code again and found that these two options are independent of each other, if one of them are not set, the grpc server will fill it with the default values, see [here](https://github.com/grpc/grpc-go/blob/b3dcc681293a029bc998fc5259252b35bfef4817/internal/transport/http2_server.go#L207). So I changed `if grpcKeepAliveInterval > time.Duration(0) && grpcKeepAliveTimeout > time.Duration(0)` to `if grpcKeepAliveInterval > time.Duration(0) || grpcKeepAliveTimeout > time.Duration(0)`.

By the way, I think the [main etcd code](https://github.com/etcd-io/etcd/blob/c623f798cbd1a61dbee202da22518ba607ad14e5/embed/etcd.go#L643) should also make this adjustment.